circuitpython_mocks/_mixins.py

```diff
@@ -1,5 +1,11 @@
 from collections import deque
-from typing import Self, Deque, Union, TYPE_CHECKING
+from typing import Deque, Union, TYPE_CHECKING
+
+try:
+    from typing import Self
+except ImportError:
+    from typing_extensions import Self
```

Comment:
These lines are missing from code coverage. Not a big deal, but we can probably forego the `try ... except` clause and only import `Self` from `typing_extensions`

Reply:
Never mind, I see typing_extensions added [`Self`](https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.Self) in 4.0.

Instead, can 
 ```suggestion
except ImportError:  # pragma: no cover
    from typing_extensions import Self
```